src/test/java/org/broadinstitute/hellbender/tools/walkers/validation/ReferenceBlockConcordanceIntegrationTest.java

```diff
@@ -0,0 +1,436 @@
+                "--" + AbstractConcordanceWalker.TRUTH_VARIANTS_LONG_NAME, truthVcf.toString(),
+                "--" + ReferenceBlockConcordance.TRUTH_BLOCK_HISTOGRAM_LONG_NAME, truthBlockHistogramFile.toString(),
+                "--" + ReferenceBlockConcordance.EVAL_BLOCK_HISTOGRAM_LONG_NAME, evalBlockHistogramFile.toString(),
+                "--" + ReferenceBlockConcordance.CONFIDENCE_CONCORDANCE_HISTOGRAM_LONG_NAME, confidenceConcordanceHistogramFile.toString(),
+        };
+        runCommandLine(args);
+
+        MetricsFile<?, String> truthBlockMetrics = new MetricsFile<>();
+        truthBlockMetrics.read(new FileReader(truthBlockHistogramFile.toFile()));
+        MetricsFile<?, String> evalBlockMetrics = new MetricsFile<>();
+        evalBlockMetrics.read(new FileReader(evalBlockHistogramFile.toFile()));
+
+        Assert.assertEquals(truthBlockMetrics.getNumHistograms(), 1);
+        Assert.assertEquals(evalBlockMetrics.getNumHistograms(), 1);
+
+        Histogram<String> truthBlockHistogram = truthBlockMetrics.getHistogram();
+        Histogram<String> evalBlockHistogram = evalBlockMetrics.getHistogram();
+
+        // Got this number by counting the <NON_REF> alt alleles in the test GVCF file
+        Assert.assertEquals(truthBlockHistogram.getSumOfValues(), 1034);
+        Assert.assertEquals(evalBlockHistogram.getSumOfValues(), 1034);
+
+        // Check block histograms both ways, in case one histogram has more entries than the other
+        truthBlockHistogram.values().forEach(bin -> {
+            Assert.assertTrue(evalBlockHistogram.containsKey(bin.getId()));
+            Assert.assertEquals(bin.getValue(), evalBlockHistogram.get(bin.getId()).getValue());
+        });
+
+        evalBlockHistogram.values().forEach(bin -> {
+            Assert.assertTrue(truthBlockHistogram.containsKey(bin.getId()));
+            Assert.assertEquals(bin.getValue(), truthBlockHistogram.get(bin.getId()).getValue());
+        });
+
+        MetricsFile<?, String> confidenceConcordanceMetrics = new MetricsFile<>();
+        confidenceConcordanceMetrics.read(new FileReader(confidenceConcordanceHistogramFile.toFile()));
+        Assert.assertEquals(confidenceConcordanceMetrics.getNumHistograms(), 1);
+        Histogram<String> confidenceConcordanceHistogram = confidenceConcordanceMetrics.getHistogram();
+
+        confidenceConcordanceHistogram.values().forEach(bin -> {
+            String[] confidenceValues = bin.getId().split(",");
+            Assert.assertEquals(confidenceValues[0], confidenceValues[1]);
+        });
+    }
+
+    private Pair<File, File> writeTestGVCFs(List<TestReferenceBlockConcordanceVariant> truthVariants, List<TestReferenceBlockConcordanceVariant> evalVariants) throws Exception {
+        File truthFile = createTempFile("truth", ".gvcf");
+        FileWriter writer = new FileWriter(truthFile);
+        writer.write("##fileformat=VCFv4.2\n");
```

Comment:
I converted it to use a GVCFWriter but I'm going to leave this comment open, because the code is much much more complicated than the primitive self-made gvcf writer. I agree that in production code we should use existing APIs, but maybe for a test it would be clearer and more fail-safe if it is actually understandable what's happening? You can make the call (once I push the changes), I'm fine either way